com.microsoft.mrtk.windowsspeech/Subsystems/TextToSpeech/WindowsTextToSpeechSubsystem.cs

```diff
@@ -0,0 +1,168 @@
+            { }
+
+            public override void Destroy()
+            {
+#if WINDOWS_UWP
+                if (synthesizer != null)
+                {
+                    synthesizer.Dispose();
+                    synthesizer = null;
+                }
+#endif // WINDOWS_UWP
+            }
+
+            #region ITextToSpeechSubsystem implementation
+
+            /// <inheritdoc/>
+            public override bool TrySpeak(string phrase, AudioSource audioSource)
+            {
+                if (audioSource == null)
+                {
+                    Debug.LogError("Must specify the AudioSource object on which the text to speech data should be applied.");
+                    return false;
+                }
+
+                if (!TrySynthesize(phrase, out byte[] waveData))
+                {
+                    return false;
+                }
+
+                // Convert from bytes to floats and create an AudioClip.
+                if (!TextToSpeechHelpers.TryConvertWaveData(
+                        waveData,
+                        out int samples,
+                        out int sampleRate,
+                        out int channels,
+                        out float[] audioFloats))
+                {
+                    Debug.LogError("Failed to convert speech audio format.");
+                    return false;
+                }
+
+                audioSource.clip = TextToSpeechHelpers.CreateAudioClip(
+                    "SynthesizedText",
+                    audioFloats,
+                    samples,
+                    channels,
+                    sampleRate);
+
+                audioSource.Play();
+                
+                return true;
+            }
+
+            /// <summary>
+            /// Attempts to synthesize the specified phrase.
+            /// </summary>
+            /// <param name="phrase">The phrase to be synthesized.</param>
+            /// <param name="waveData">Receives the audio (wave) data upon successful synthesis.</param>
+            /// <returns>True if the phrase was successfully synthesized, or false.</returns>
+            private bool TrySynthesize(string phrase, out byte[] waveData)
+            {
+                waveData = null;
+
+                if (string.IsNullOrWhiteSpace(phrase))
+                {
+                    Debug.LogWarning("Nothing to speek");
+                    return false;
+                }
+
+#if WINDOWS_UWP
+                // By specifying to use the default synthesizer voice, we will speak in
+                // the voice selected, in Settings, by the user.
+                synthesizer.Voice = SpeechSynthesizer.DefaultVoice;
+
+                Task<SpeechSynthesisStream> synthTask = synthesizer.SynthesizeTextToStreamAsync(phrase).AsTask<SpeechSynthesisStream>();
+                while (!synthTask.IsCompleted)
+                {
+                    System.Threading.Thread.Sleep(0);
```

Comment:
Since I assume this is running on Unity's main thread, this will be blocking, right? Or am I misreading?

If it's blocking, is there a reason we wouldn't want to use async/await here to fix that, instead of sleep?

Reply:
which, depending on where customers place the call, may require them to decorate unity functions

Reply:
just noticed this is a private method (thx @keveleigh!). will change

Reply:
just revisited this. adding awaits will require a #pragma for the non-UWP scenario _AND_ it has been pointed out that the pinvoke call is also blocking.

i am going to wrap the pinvoke in a Task.Run

Reply:
^^^ and add a comment on why the pattern was used

Reply:
Actually, ignore all of the above.... 😆 looking into a better solution

Reply:
done :)